libretroshare/src/gxs/rsdataservice.h

```diff
@@ -38,54 +38,45 @@ class MsgUpdate
 template<class ID, class MetaDataClass> class t_MetaDataCache
 {
 public:
-    t_MetaDataCache() : mCache_ContainsAllMetas(false) {}
-    virtual ~t_MetaDataCache()
-    {
-        for(auto it: mMetas)
-            delete it.second;
-
-        for(auto it: mOldCachedItems)
-            delete it.second;
-    }
+    t_MetaDataCache()
+        : mCache_ContainsAllMetas(false)
+    {}
+    virtual ~t_MetaDataCache() = default;
 
     bool isCacheUpToDate() const { return mCache_ContainsAllMetas ; }
     void setCacheUpToDate(bool b) { mCache_ContainsAllMetas = b; }
 
-	void getFullMetaList(std::map<ID,MetaDataClass*>& mp) const { mp = mMetas ; }
-    void getFullMetaList(std::vector<const MetaDataClass*>& mp) const { for(auto& m:mMetas) mp.push_back(m.second) ; }
+    void getFullMetaList(std::map<ID,std::shared_ptr<MetaDataClass> >& mp) const { mp = mMetas ; }
+    void getFullMetaList(std::vector<std::shared_ptr<MetaDataClass> >& mp) const { for(auto& m:mMetas) mp.push_back(m.second) ; }

```

Comment:
If relevant shared_ptr can point to a const object like this `std::shared_ptr<const MetaDataClass>`